### PR TITLE
Updates and clarifications in deployment models doc

### PR DESCRIPTION
**Pull request by frankbu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/en/docs/ops/deployment/deployment-models/index.md [hunk 1]
```diff
@@ -123,43 +120,44 @@ You can configure inter-cluster communication based on the
...
 To ensure that DNS lookup succeeds, you must deploy a Kubernetes `Service` to
 each cluster that consumes that service. This ensures that regardless of
 where the request originates, it will pass DNS lookup and be handed to Istio
 for proper routing.
-
 This can also be achieved with Istio `ServiceEntry`, rather than Kubernetes
-`Service`. `ServiceEntry` does not configure DNS, however. This means that
-DNS will have to be configured either manually or with automated tooling,
-such as the [Istio CoreDNS Plugin](https://github.com/istio-ecosystem/istio-coredns-plugin).
+`Service`. However, a `ServiceEntry` does not configure the Kubernetes DNS server
```
Comment (nmittler) [inline content/en/docs/ops/deployment/deployment-models/index.md:149]: I'm not sure that the new wording in this paragraph is an improvement. The sentence seems like a bit long-winded.

File: content/en/docs/ops/deployment/deployment-models/index.md [hunk 2]
```diff
@@ -172,9 +170,6 @@ There are a few efforts in progress that will help simplify the DNS story:
   Kubernetes. There is also work in progress to build layers of `MCS` support
   into Istio, which would allow Istio to work with any cloud vendor `MCS`
   controller or even act as the `MCS` controller for the entire mesh.
-
-While `Admiral` is available today, the Istio and Kubernetes communities are
```
Comment (nmittler) [inline content/en/docs/ops/deployment/deployment-models/index.md:176]: Why remove this last sentence and not just update it?

File: content/en/docs/ops/deployment/deployment-models/index.md [hunk 3]
```diff
@@ -330,21 +320,17 @@ clusters, zones, or regions.
...
+  rollout, for example, to canary configuration changes in a sub-section of the mesh
+  controlled by a given primary cluster.
 
-- Selective Service Visibility: You can choose which `Service`/`ServiceEntry`
-  resources to make visible to each control plane. This can be used to
-  establish service-level isolation between clusters. For example, an
-  administrator may choose to deploy the `HelloWorld` service to Cluster A,
-  but not Cluster B. Any attempt to call `HelloWorld` from Cluster B will
-  fail the DNS lookup.
+- Selective service visibility: You can restrict service visibility to part
```
Comment (nmittler) [inline content/en/docs/ops/deployment/deployment-models/index.md:332]: Why did you remove the example in this bullet?

File: content/en/docs/ops/deployment/deployment-models/index.md [hunk 4]
```diff
@@ -330,21 +320,17 @@ clusters, zones, or regions.
...
+  the outage is limited to only workloads in clusters managed by that control plane.
 
 - Configuration isolation: You can make configuration changes in one cluster,
   zone, or region without impacting others.
 
-- Controlled Rollout: You have more fine-grained control over configuration
-  rollout (e.g. one cluster at a time). For large meshes with many clusters,
-  this process can be automated with CI/CD pipelines.
+- Controlled rollout: You have more fine-grained control over configuration
+  rollout, for example, to canary configuration changes in a sub-section of the mesh
```
Comment (nmittler) [inline content/en/docs/ops/deployment/deployment-models/index.md:329]: The new wording seems less clear to me.

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 2 above)
Comment (frankbu) [reply content/en/docs/ops/deployment/deployment-models/index.md:176]: My reasoning is that the above sentences already make it clear that both Istio and Kubernetes are actively building solutions, and I don't think we really want to over hype Admiral, since it's not really a great solution.

File: content/en/docs/ops/deployment/deployment-models/index.md [hunk 5]
```diff
@@ -266,33 +261,10 @@ cluster.
...
-user workloads but no control plane. Remote clusters rely on a control plane
-residing outside of the cluster (e.g. in a primary cluster). In this case, the
-control plane must be made accessible to the remote cluster.
-
-This can be achieved by exposing the control plane through an Istio Gateway.
-Cloud vendors may provide options, such as internal load balancers, for
-providing this sort of capability without exposing the control plane on the
-public internet. Such an option, if it exists and meets your requirements,
-will likely be the best choice.
+Multicluster deployments can also share control plane instances. In this case,
```
Comment (nmittler) [inline content/en/docs/ops/deployment/deployment-models/index.md:264]: This paragraph now seems like it's talking about two separate things at once (primary and remote). The old wording intentionally tried to describe in entirety the concept of primary clusters before mentioning remotes. I think that's a better approach in general, since it stays on topic. Perhaps subsections would help, where we first talk about primaries and then remotes?

Comment (frankbu) [reply content/en/docs/ops/deployment/deployment-models/index.md:264]: That was intentional. The section is talking control planes in multicluster configurations, which are comprised of two kinds of clusters. First define the terms with the pictures, then talk about how they work. 

I had considered splitting the rest (after the overview) into separate sections, but concluded there wasn't really enough material to justify it.

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 3 above)
Comment (frankbu) [reply content/en/docs/ops/deployment/deployment-models/index.md:332]: I thought its TMI for this doc, but can add it back if you feel strongly about it.

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 4 above)
Comment (frankbu) [reply content/en/docs/ops/deployment/deployment-models/index.md:329]: I was trying to eliminate duplication, it talked about the same things here and earlier, so I combined them. Maybe it just needs a little more detail now? How about:
```suggestion
  rollout (e.g., one cluster at a time). You can also canary configuration changes in a sub-section of the mesh
```

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 1 above)
Comment (frankbu) [reply content/en/docs/ops/deployment/deployment-models/index.md]: I guess that's true, I split it into two sentences again.

File: content/en/docs/ops/deployment/deployment-models/index.md [hunk 6]
```diff
@@ -301,6 +273,24 @@ will likely be the best choice.
     caption="A service mesh with a primary and a remote cluster"
     >}}
 
+To support remote clusters in a multicluster mesh, the control plane in
+a primary cluster must be made accessible to them. For clusters spanning networks,
```
Comment (nmittler) [inline content/en/docs/ops/deployment/deployment-models/index.md:277]: Accessibility is only true for multi-network. For single network, it's about having a load balanced IP address for the control plane (so we don't have to use pod IP which is not stable).

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 4 above)
Comment (nmittler) [reply content/en/docs/ops/deployment/deployment-models/index.md:329]: Sure that works.

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 5 above)
Comment (nmittler) [reply content/en/docs/ops/deployment/deployment-models/index.md:264]: I actually think it might be worth it. The remote section can talk about how to connect to a primary. Maybe give it a try and see if it works?

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 3 above)
Comment (nmittler) [reply content/en/docs/ops/deployment/deployment-models/index.md:332]: It helps to illustrate what we mean by `restricting service visibility`, I think it's probably useful.

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 2 above)
Comment (nmittler) [reply content/en/docs/ops/deployment/deployment-models/index.md:176]: ok I'm fine removing it.

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 6 above)
Comment (frankbu) [inline content/en/docs/ops/deployment/deployment-models/index.md:277]: How about:
```suggestion
a primary cluster must be accessable via a stable IP (e.g., a cluster IP). For clusters spanning networks,
```

Comment (nmittler) [reply content/en/docs/ops/deployment/deployment-models/index.md:277]: lgtm

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 5 above)
Comment (frankbu) [reply content/en/docs/ops/deployment/deployment-models/index.md:264]: Looking at it, this section won't flow well with subsections. Let's leave it as is for now. We can reconsider next time we reorg this doc 😄

File: content/en/docs/ops/deployment/deployment-models/index.md (hunk 3 above)
Comment (frankbu) [reply content/en/docs/ops/deployment/deployment-models/index.md:332]: added

**Review by ericvn — APPROVED** [review]
